### DISCUSSION TITLE
FRITZ!Box 6660 Cable not responding to FTP

### DISCUSSION THREAD

**Discussion by maexbower** [Freetz-NG/freetz-ng#641] [Allgemein] [answered]
I have a new FRITZ!Box 6660 Cable with the original Firmware Version 07.29 on it. 
I can build the Feetz-NG Image without any issues but when I try to restart the Fritz!Box to flash the Image via FTP it does not work.
push_firmware simply fails with: Connection refused
I tried it now with the following setups:
Windows 11, push_firmware running with WSL (Windows Subsystem for LInux) connected via LAN (powerLAN in between)
Windows10 with Ubuntu 22.04 in Hyper V. Exclusivly connected to a LAN port and directly with a cable connected to the Fritz!Box. 
I flashed again the original firmware and did a factory reset of the Fritz!Box.

When I try to connect manually with the ftp command to the fritz box as soon as it replies to the ping again, it as well just says connection refused (tried for 20 seconds each second).

Someone had similar issues? Am I just to dumb to do it?

**Comment by maexbower** [comment, marked as answer]
Update after desperate try and error: the timing functions in the script do not work in my setup. If I use the flags -w -f to force the push_firmware script to directly push when I hit enter, I was able to catch exactly the right moment. 
interestingly all tests where I tried to connect within a while loop each second we’re failing. In my case it was working the best when I watched the activity LED on my LAN Port.

**Comment by AkteX** [comment]
You can use recovery images from other fritzboxes to stop your 6660 in the bootloader and use adam2-ftp afterwards. I am using FRITZ.Box_7590-07.29-recover.exe for this purpose.

**Comment by fda77** [comment]
The timing of the ping + connect depends on many things like docker/VM/native linux distro, network switched etc
Someone should devel an simple Java program
